Externals/hidapi/CMakeLists.txt

```diff
@@ -0,0 +1,24 @@
+project(hidapi)
+
+set(SRCS)
+set(LIBS)
+
+if(APPLE)
+  list(APPEND SRCS mac/hid.c)
+elseif(MSVC)
+  list(APPEND SRCS windows/hid.c)
+else()
+  find_package(Libudev)
+  if(${CMAKE_SYSTEM_NAME} STREQUAL "Linux" AND LIBUDEV_FOUND)
+    list(APPEND SRCS linux/hid.c)
+    list(APPEND LIBS udev)
+  else()
+    list(APPEND SRCS libusb/hid.c)
+  endif()
+endif()
+
+add_library(hidapi ${SRCS})
+target_link_libraries(hidapi PRIVATE ${LIBS})
+target_include_directories(hidapi PUBLIC hidapi)
+
+add_library(Hidapi::Hidapi ALIAS hidapi)
```

Comment:
Out of curiosity, what's the main usage-case of `ALIAS` targets? I get that they prevent clashing, but CMake's docs don't really do a very good job of explaining when to use them, nor is there an example of a simple problem solved by them. (@Orphis as well, if you want to drop knowledge on me).